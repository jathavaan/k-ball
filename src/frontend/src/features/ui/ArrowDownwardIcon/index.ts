export * from "./ArrowDownwardIcon.style.ts";
export * from "./ArrowDownwardIcon.types.ts";
export * from "./ArrowDownwardIcon.tsx";
