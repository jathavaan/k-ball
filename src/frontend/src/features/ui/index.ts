export * from "./card-header/CardHeader";
export * from "./card-content-text/CardContentText";
export * from "./error-alert/ErrorAlert";
export * from "./linear-progress-bar/LinearProgressBar";
export * from "./image-container/ImageContainer";
export * from "./select/Select";
