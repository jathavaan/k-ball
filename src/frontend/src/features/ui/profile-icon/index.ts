// index.ts in profileIcon folder
export * from "./profileIcon.api";
export * from "./profileIcon.hooks";
export * from "./profileIcon.types";
