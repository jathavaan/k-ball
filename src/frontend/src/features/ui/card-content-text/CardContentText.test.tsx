import { render, screen } from "@testing-library/react";
import { CardContentText } from "@features/ui";

describe("CardContentText", () => {
  it("Should render the title and text", () => {
    const mockProps = {
      title: "Test Title",
      text: "Test Text",
    };

    render(<CardContentText {...mockProps} />);

    expect(screen.getByText(`${mockProps.title}:`)).toBeDefined();
    expect(screen.getByText(mockProps.text)).toBeDefined();
  });

  it("Should match CardContentText snapshot", () => {
    const mockProps = {
      title: "Test Title",
      text: "Test Text",
    };

    const { asFragment } = render(<CardContentText {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
