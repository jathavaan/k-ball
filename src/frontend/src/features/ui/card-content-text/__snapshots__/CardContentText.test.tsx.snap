// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardContentText > Should match CardContentText snapshot 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-x09oxf-MuiTypography-root"
    text="Test Text"
    title="Test Title"
  >
    <b>
      Test Title:
    </b>
     Test Text
  </p>
</DocumentFragment>
`;

exports[`CardContentText > Should match CardContentText snapshot 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-x09oxf-MuiTypography-root"
    text="Test Text"
    title="Test Title"
  >
    <b>
      Test Title:
    </b>
     Test Text
  </p>
</DocumentFragment>
`;
